### PR TITLE
feat(health): add event-loop health check

### PR DESCRIPTION
**Pull request by s4heid**

### CODE REVIEW

File: src/main/java/io/neonbee/health/EventLoopHealthCheck.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            return 0;
+        }
+
+        AtomicInteger pendingTasks = new AtomicInteger(0);
+        neonBee.getVertx().nettyEventLoopGroup().forEach(elg -> {
+            SingleThreadEventExecutor singleEventExecutor = ((SingleThreadEventExecutor) elg);
+            if (singleEventExecutor.pendingTasks() > 0) {
+                pendingTasks.getAndAdd(singleEventExecutor.pendingTasks());
+                if (LOGGER.isInfoEnabled()) {
+                    LOGGER.info("[{}] Pending Tasks of \"{}\": {}", neonBee.getNodeId(),
```
Comment (s4heid) [inline src/main/java/io/neonbee/health/EventLoopHealthCheck.java:79]: should be debug

Comment (pk-work) [reply src/main/java/io/neonbee/health/EventLoopHealthCheck.java:79]: trace

**Review by pk-work — CHANGES_REQUESTED** [review]

File: src/main/java/io/neonbee/health/EventLoopHealthCheck.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * Name of the health check.
+     */
+    public static final String NAME = "eventloop";
+
+    @VisibleForTesting
+    static final String CRITICAL_EVENTBUS_SIZE_KEY = "criticalEventLoopSize";
+
+    private static final int DEFAULT_CRITICAL_EVENTBUS_SIZE = 1;
```
[the remark this answers is not in the document]
Comment (s4heid) [reply src/main/java/io/neonbee/health/EventLoopHealthCheck.java:31]: Changed to 5. We can adapt this once we have more experience with it.

File: src/main/java/io/neonbee/health/EventLoopHealthCheck.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        };
+    }
+
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    private int eventBusSize(NeonBee neonBee) {
+        Context context = neonBee.getVertx().getOrCreateContext();
+        if (!context.isEventLoopContext()) {
+            LOGGER.warn("Cannot perform Event Loop Health Check. Context not attached to Event Loop.");
+            return 0;
+        }
```
[the remark this answers is not in the document]
Comment (s4heid) [reply src/main/java/io/neonbee/health/EventLoopHealthCheck.java:71]: Agreed

File: src/test/java/io/neonbee/internal/verticle/HealthCheckVerticleTest.java [hunk 4]
```diff
@@ -41,7 +41,6 @@ void testRetrieveData(VertxTestContext testContext) {
         Future<JsonArray> response = requestData(req);
 
         assertData(response, resp -> {
-            assertThat(resp.size()).isEqualTo(1);
```
[the remark this answers is not in the document]
Comment (s4heid) [reply src/test/java/io/neonbee/internal/verticle/HealthCheckVerticleTest.java:44]: As discussed, this is already checked elsewhere and needs to be adapted every single time if a new health check is added. Thus, removing it.

**Review by pk-work — APPROVED** [review]

**Pull request merged**